oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationEndpointConfigurer.java

```diff
@@ -83,6 +84,7 @@ public OAuth2AuthorizationEndpointConfigurer authorizationRequestConverter(Authe
 	 * @return the {@link OAuth2AuthorizationEndpointConfigurer} for further configuration
 	 */
 	public OAuth2AuthorizationEndpointConfigurer authenticationProvider(AuthenticationProvider authenticationProvider) {
+		Assert.notNull(authenticationProvider, "AuthenticationProvider cannot be null");
```

Comment:
Use exact parameter name in assertions -> "authenticationProvider". Please update other assertions.